### DISCUSSION TITLE
YunoHost

### DISCUSSION THREAD

**Discussion by ianalexis** [tteck/Proxmox#898] [Ideas]
YunoHost is an operating system aiming for the simplest administration of a server, and therefore democratize [self-hosting](https://yunohost.org/en/selfhosting), while making sure it stays reliable, secure, ethical and lightweight. It is a copylefted libre software project maintained exclusively by volunteers. Technically, it can be seen as a distribution based on [Debian GNU/Linux](https://debian.org/) and can be installed on [many kinds of hardware](https://yunohost.org/en/install).

Install on debian 11 with https://yunohost.org/en/install/hardware:vps_debian

**Comment by tteck** [comment]
Here's something for you to test

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/yunohost-v5.sh)"
```
Please let me know how it goes (I've never used YunoHost)

**Comment by ianalexis** [comment]
i think is not working (or ar least its taking a really long time...).
I think its better to add a curl install first an then accept the promps (always needed to accept).

First promp:
`Caution !
Your configuration files for :
- postfix
- dovecot
- mysql
- nginx
- metronome
will be overwritten !
Are you sure you want  to proceed with the installation of Yunohost?`

Response: YES

Second Promp:
`To improve the security of your server, it is recommended to let YunoHost
manage the SSH configuration.
Your current SSH configuration differs from the recommended configuration.
If you let YunoHost reconfigure it, the way you connect to your server
through SSH will change in the following way:
- you will not be able to connect as root through SSH. Instead you should
use the admin user ;
Do you agree to let YunoHost apply those changes to your configuration and
therefore affect the way you connect through SSH ?`

Response: YES

**Reply by tteck** [reply]
It takes a really long time to install
It's an unattended install (no prompts)
Use the "Verbose Mode" to watch the progress

**Reply by ianalexis** [reply]
Working perfect,
Only need a `apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg' > /dev/null 2>&1` after install because after the yuno installer it gets "The repository 'https://packages.sury.org/php bullseye InRelease' is not signed." (Yunohost error, surely they are going to fixit soon).

YunoHost is like an "easy" installer with user management, email, HASSIO etc...
IMHO, its the easiest way to install nextcloud with onlyoffice and other services quick and easy **¡¡¡in a LXC!!!.**

Some applications already exist in your repository and others are frequently requested by other users here. The only “downside” it’s the need from the community to upgrade the packages so it gets a little delay.

You could see all the apps here: https://yunohost.org/en/apps

Thanks Tteck! Sooner than later, I will buy you some coffes!

**Reply by tteck** [reply]
Do I need to add `apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg'` to the script?

**Reply by ianalexis** [reply]
Nah, maybe just a comment in the description like "If you can't update or you have the `The repository 'https://packages.sury.org/php bullseye InRelease' is not signed.` error, run: `apt-key adv --fetch-keys 'https://packages.sury.org/php/apt.gpg' > /dev/null 2>&1`.

**Reply by tteck** [reply]
Easier to just add it to the script

**Reply by OddSquirrel** [reply]
Dang, that's certainly a nice project, wasn't even aware of it – thanks to both of you! 😊

**Reply by ianalexis** [reply]
but if they fix it? couldnt it break something?

**Reply by Vakrehus** [reply]
```
apt-get install apt-transport-https lsb-release ca-certificates
wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
```
this is needed for working updates.

**EDIT:** here https://github.com/tteck/Proxmox/discussions/898#discussioncomment-4615418

**Reply by tteck** [reply]
Done

**Reply by ianalexis** [reply]
Now failing:

> + wget -q -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
> ++ EXIT=4
> ++ LINE=93
> ++ error_exit
> ++ trap - ERR
> ++ local 'reason=Unknown failure occurred.'
> ++ local 'msg=Unknown failure occurred.'
> ++ local 'flag=\033[01;31m‼ ERROR \033[m4@93'
> ++ echo -e '\033[01;31m‼ ERROR \033[m4@93 Unknown failure occurred.'
> ‼ ERROR 4@93 Unknown failure occurred.
> ++ exit 4
> + exit

**Comment by Vakrehus** [comment]
do this line
```
apt-key adv --fetch-keys ‘https://packages.sury.org/php/apt.gpg’ > /dev/null 2>&1
```
before
```
apt-get install apt-transport-https lsb-release ca-certificates
wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg
```

**Comment by johndoe0815** [comment]
While the yunohost container works for me, including update, backup of the container in proxmox does not work. This is the only container where the proxmox backup functionality consistently throws an error. I have restarted the container, I have tried the backup in all three modes, it always fails. 
Does someone have a clue, what this could be?

The output of the backup process is:

> INFO: create storage snapshot 'vzdump'
  Logical volume "snap_vm-112-disk-0_vzdump" created.
INFO: creating vzdump archive '/var/lib/vz/dump/vzdump-lxc-112-2023_02_16-10_58_21.tar.zst'
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-@types-q-1.5.5-75a2a8e7d8ab4b230414505d92335d1dcb53a6df-integrity/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-estraverse-4.3.0-398ad3f3c5a24948be7725e83d11a7de28cdbd1d-integrity/node_modules/estraverse/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-color-name-1.1.3-a7d0558bd89c42f795dd42328f740831ca53bc25-integrity/node_modules/color-name/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-has-property-descriptors-1.0.0-610708600606d36961ed04c196193b6a607fa861-integrity/node_modules/has-property-descriptors/.github/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-estraverse-5.3.0-2eea5290702f26ab8fe5370370ff86c965d21123-integrity/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-get-intrinsic-1.1.1-15f59f376f855c446963948f0d24cd3637b4abc6-integrity/node_modules/get-intrinsic/.github/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-get-intrinsic-1.1.1-15f59f376f855c446963948f0d24cd3637b4abc6-integrity/node_modules/get-intrinsic/test/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-coa-2.0.2-43f6c21151b4ef2bf57187db0d73de229e3e7ec3-integrity/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-@babel-eslint-parser-7.19.1-4f68f6b0825489e00a24b41b6a1ae35414ecd2f4-integrity/node_modules/: Cannot savedir: Bad message
INFO: tar: ./usr/local/share/.cache/yarn/v6/npm-@kevinfaguiar-vue-twemoji-picker-5.7.4-76dd0a2dc9cc350af0886eae985492c868b02e49-integrity/node_modules/@kevinfaguiar/: Cannot savedir: Bad message
INFO: Total bytes written: 5931663360 (5.6GiB, 46MiB/s)
INFO: tar: Exiting with failure status due to previous errors
INFO: cleanup temporary 'vzdump' snapshot
  Logical volume "snap_vm-112-disk-0_vzdump" successfully removed
ERROR: Backup of VM 112 failed - command 'set -o pipefail && lxc-usernsexec -m u:0:100000:65536 -m g:0:100000:65536 -- tar cpf - --totals --one-file-system -p --sparse --numeric-owner --acls --xattrs '--xattrs-include=user.*' '--xattrs-include=security.capability' '--warning=no-file-ignored' '--warning=no-xattr-write' --one-file-system '--warning=no-file-ignored' '--directory=/var/lib/vz/dump/vzdump-lxc-112-2023_02_16-10_58_21.tmp' ./etc/vzdump/pct.conf ./etc/vzdump/pct.fw '--directory=/mnt/vzsnap0' --no-anchored '--exclude=lost+found' --anchored '--exclude=./tmp/?*' '--exclude=./var/tmp/?*' '--exclude=./var/run/?*.pid' ./ | zstd --rsyncable '--threads=1' >/var/lib/vz/dump/vzdump-lxc-112-2023_02_16-10_58_21.tar.dat' failed: exit code 2
INFO: Failed at 2023-02-16 11:00:28
INFO: Backup job finished with errors

**Reply by ianalexis** [reply]
Its working fine for me. Did you tried reinstalling?
